core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -113,7 +114,28 @@ public synchronized void registerFilter(RestFilter preProcessor) {
     }
 
     /**
-     * Registers a rest handler to be execute when the provided method and path match the request.
+     * Registers a REST handler to be executed when the provided {@code method} and {@code path} match the request.
+     *
+     * @param method GET, POST, etc.
+     * @param path Path to handle (e.g., "/{index}/{type}/_bulk")
+     * @param handler The handler to actually execute
+     * @param deprecationMessage The message to log and send as a header in the response
+     * @param logger The existing deprecation logger to use
+     */
+    public void registerHandlerAsDeprecationHandler(RestRequest.Method method, String path, RestHandler handler,
```

Comment:
There is a type called `DeprecationRestHandler`. That's what handlers passed to this method get wrapped into. But you don't and can't pass one of those to this method yet that's what the name `registerDeprecatedHandler` suggests you can and maybe should do. The method does not register deprecated handlers, it registers plain handlers as deprecated handlers. I think `registerAsDeprecatedHandler` is fine though. :smile: